zrml/prediction-markets/src/lib.rs

```diff
@@ -443,11 +445,27 @@ mod pallet {
                     Error::<T>::MarketEditRequestAlreadyInProgress
                 );
                 m.status = new_status;
+
+                if m.is_redeemable() {
+                    for outcome in m.outcome_assets(market_id) {
+                        let admin = Self::market_account(market_id);
+                        let is_sufficient = true;
+                        let min_balance = 1u8;
+                        T::AssetCreator::create(
+                            outcome.into(),
+                            admin,
+                            is_sufficient,
+                            min_balance.into(),
+                        )?;
+                    }
+                }
+
                 Ok(())
             })?;
 
             Self::unreserve_creation_bond(&market_id)?;
 
+            T::OnStateTransition::on_activation(&market_id).result?;
```

Comment:
`OnStateTransition` is set to `()` in the mock, meaning this behavior is not tested at all. I think this is a good point to start using mocks for testing (let's not use zrml-parimutuel to test this!). But the behavior should definitely be tested before going into production.

Reply:
I think it's sufficient to have tests for the underlying `MarketTransitionApi` that is invoked, as done at https://github.com/zeitgeistpm/zeitgeist/pull/1281/commits/e7563f7a4dfbec3249fcd42705ca5fd798cc1136 . What else should be tested here besides that?

Reply:
I would add a test that verifies that the `MarketTransitionApi` is actually called. Right now, I'm assuming I can delete that line and the tests would all pass due to the configuration of the mock.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1282/commits/0cdac69fe7d5b91af09a302a7b01753569f7213d